Code/GraphMol/Deprotect/Deprotect.cpp

```
@@ -0,0 +1,107 @@
+     {"alcohol", "[O;R0:1][C;R0](=O)[C;H3]>>[O:1]", "Ac", "acetyl"},
+     {"alcohol", "[c;H1]1[c;H1]c(O[C;H3])[c;H1][c;H1]c1[C;H2][O;D2&R0:1]>>[O;H1:1]", "PMB", "para-methoxybenzyl_ether"},
+     {"alcohol", "c1ccc(C([NX3;H0,H1,H2:1])(c2ccccc2)c2ccccc2)cc1>>[N:1]", "Tr", "triphenylmethyl"},
+     {"amine", "O=C1[N;H0:1]C(c2[c;H1][c;H1][c;H1][c;H1]c21)=O>>[N:1]", "Phth", "phthaloyl"},
+     {"amine", "[#7:1]COCC[Si]([C;R0])([C;R0])[C;R0]>>[#7:1]", "SEM", "trimethylsilylethoxymethyl"},
+     {"amine", "[C;H3;R0][O;R0]c1[c;H1][c;H1][c]([NX3;H0,H1;!$(NC=O):1])[c;H1][c;H1]1>>[N:1]", "PMP", "para-methoxyphenyl"},
+     {"amine", "[C;H3]c1[c;H1][c;H1]c(S(=O)(=O)[NX3;H0,H1;!$(NC=O):1])[c;H1][c;H1]1>>[N:1]", "Ts", "tosyl"},
+     {"amine", "[C;R0][C;R0]([C;R0])([O;R0][C;R0](=[O;R0])[NX3;H0,H1:1])C>>[N:1]", "Boc", "tert-butyloxycarbonyl"},
+     {"amine", "[N;H0,H1:1]C(=O)C(F)(F)F>>[N:1]", "TFA", "trifluoroacetyl"},
+     {"amine", "[NX3;H0,H1;!$(NC=O):1]C(=O)c1[c;H1][c;H1][c;H1][c;H1][c;H1]1>>[N:1]", "Bz", "benzoyl"},
+     {"amine", "[NX3;H0,H1;!$(NC=O):1]C(OCC1C(cccc2)c2c3c1cccc3)=O>>[N:1]", "Fmoc", "9-fluorenylmethyloxycarbonyl"},
+     {"amine", "[NX3;H0,H1;!$(NC=O):1][C;H2]c1[c;H1][c;H1][c;H1][c;H1][c;H1]1>>[N:1]", "Bn", "benzyl"},
+     {"amine", "[NX3;H0,H1;!$(NC=O):1][C;R0](=O)[C;H3]>>[N:1]", "Ac", "acetamide"},
+     {"amine", "[NX3;H0,H1;!$(NC=O):1][C;R0](=O)[O;R0][C;R0]c1[c;H1][c;H1][c;H1][c;H1][c;H1]1>>[N:1]", "Cbz", "carbobenzyloxy"}
+  };
+  return deprotections;
+}
+  
+std::unique_ptr<ROMol> deprotect(const ROMol& mol, const std::vector<DeprotectData> &deprotections) {
+  std::set<std::string> found;
+  std::vector<std::string> deprotections_used;
+  
+  bool something_happened = true;
+  auto m = boost::make_shared<ROMol>(mol);
+  while (something_happened) {
+    something_happened = false;
+    for(auto & deprotect : deprotections) {
+      if (!deprotect.isValid()) {
+	// error and contine;
```

Nit: you don't actually "error" here

---

That was just a comment.  Should I error?  Should I throw an error like warning:  skipping invalid deprotection....

---

It's just a nit that the comment isn't right.
I think it's fine to not actually generate an error and I wouldn't suggest doing any warnings here... maybe when actually constructing the deprotections?